src/Libraries/PythonNodeModelsWpf/IronPythonCompletionProvider.cs

```diff
@@ -64,28 +60,60 @@ public ScriptScope Scope
 
         /// <summary>
         /// Maps a regex to a particular python type.  Useful for matching things like dicts,
-        /// floats, strings, etc.  Initialized by 
+        /// floats, strings, etc.  Initialized by
         /// </summary>
         public Dictionary<string, Type> RegexToType = new Dictionary<string, Type>();
+        
+        /// <summary>
+        /// Maps a basic variable regex to a basic python type.
+        /// </summary>
+        public List<Tuple<Regex, Type>> BasicVariableTypes;
+        
+        /// <summary>
+        /// Tracks already referenced CLR modules
+        /// </summary>
+        public HashSet<string> clrModules { get; set; }
+        
+        /// <summary>
+        /// Keeps track of failed statements to avoid poluting the log
+        /// </summary>
+        public Dictionary<string, int> badStatements { get; set; }
 
         /// <summary>
         /// A bunch of regexes for use in introspaction
         /// </summary>
-        public static string commaDelimitedVariableNamesRegex = @"(([0-9a-zA-Z_]+,?\s*)+)";
-        public static string variableName = @"([0-9a-zA-Z_]+(\.[a-zA-Z_0-9]+)*)";   
-        public static string doubleQuoteStringRegex = "(\"[^\"]*\")";
-        public static string singleQuoteStringRegex = "(\'[^\']*\')";
+        public static string commaDelimitedVariableNamesRegex = @"(([0-9a-zA-Z_]+,?\s?)+)";
+        public static string variableName = @"([0-9a-zA-Z_]+(\.[a-zA-Z_0-9]+)*)";
+        public static string quotesStringRegex = "[\"']([^\"']*)[\"']";
```

Comment:
I think it would be very useful to add comments/summaries to some of these - especially if they are public to describe anything that is not obvious from the field names.